### ISSUE TITLE
GP ungracefully fails handling a single choice in search space

### ISSUE THREAD

**Issue by synapticarbors** [msmbuilder/osprey#148]
When using GP, if you have an enum with a single choice, the GP strategy fails with the following message:

```
Traceback (most recent call last):
  File "/Users/zzz/anaconda/envs/osprey/bin/osprey", line 9, in <module>
    load_entry_point('osprey', 'console_scripts', 'osprey')()
  File "/Users/zzz/Projects/github/osprey/osprey/cli/main.py", line 37, in main
    args_func(args, p)
  File "/Users/zzz/Projects/github/osprey/osprey/cli/main.py", line 42, in args_func
    args.func(args, p)
  File "/Users/zzz/Projects/github/osprey/osprey/cli/parser_worker.py", line 8, in func
    execute(args, parser)
  File "/Users/zzz/Projects/github/osprey/osprey/execute_worker.py", line 77, in execute
    sessionbuilder=config.trialscontext)
  File "/Users/zzz/Projects/github/osprey/osprey/execute_worker.py", line 102, in initialize_trial
    params = strategy.suggest(history, searchspace)
  File "/Users/zzz/Projects/github/osprey/osprey/strategies.py", line 276, in suggest
    X, Y, V, ignore = self._get_data(history, searchspace)
  File "/Users/zzz/Projects/github/osprey/osprey/strategies.py", line 232, in _get_data
    point = searchspace.point_to_gp(param_dict)
  File "/Users/zzz/Projects/github/osprey/osprey/search_space.py", line 112, in point_to_gp
    return [var.point_to_gp(point_dict[var.name]) for var in self]
  File "/Users/zzz/Projects/github/osprey/osprey/search_space.py", line 234, in point_to_gp
    return float(index) / (len(self.choices) - 1)
ZeroDivisionError: float division by zero
```

From the config:

```
search_space:
  # the search space is specified by listing the variables you want to
  # optimize over and their bounds for float and int typed variables,
  # or the possible choices for enumeration-typed variables.
  featurizer__types:
    choices:
      - ['phi', 'psi']
      #- ['phi', 'psi', 'chi1']
    type: enum
```

**Comment by brookehus** [comment]
Thanks for pointing this out- the error message should be more helpful. 

As a workaround, if you are restricting any part of your model to a single choice, you can initialize it that way at the beginning of your pipeline and remove it from your search space, e.g. for the types of the dihedral featurizer,
```
estimator:
    eval: |
        Pipeline([
                ('featurizer', DihedralFeaturizer(types=['phi', 'psi'], sincos=True)),
                ('cluster', MiniBatchKMeans()), 
                ('msm', MarkovStateModel(n_timescales=5, verbose=False)),
                                    ])
```

As a little bit of a hack when I have wanted to create a model that isn't actually searching anything I would do something like:

```
search_space:
  featurizer__types:
    choices:
      - ['phi', 'psi']
      - ['phi', 'psi']
    type: enum
```

**Comment by cxhernandez** [comment]
fixed in #191
